nixos/modules/services/x11/xidlehook.nix

```
@@ -0,0 +1,145 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.xserver.xidlehook;
+in
+  {
+    options = {
+      services.xserver.xidlehook = {
+        enable = mkEnableOption "xidlehook";
+
+        notWhenFullscreen = mkOption {
+          default = true;
+          description = "Whether to lock the screen when an application is fullscreened.";
+          type = types.bool;
+        };
+
+        notWhenAudio = mkOption {
+          default = false;
+          description = "Whether to lock the screen when audio is playing.";
+          type = types.bool;
+        };
+
+        socket = mkOption {
+          default = ".local/xidlehook.socket";
+          description = ''
+            Socket file for controlling the daemon. The path is relative to the
+            user's home directory. Set to null to not have one.
+          '';
+          type = types.nullOr types.str;
+        };
+
+        timerPkgs = mkOption {
+          default = with pkgs; [xorg.xrandr gawk i3lock-fancy-rapid];
```

1. why Gawk?
2. whitespace.
```suggestion
          default = with pkgs; [ xorg.xrandr gawk i3lock-fancy-rapid ];
```